### DISCUSSION TITLE
How to check if the current channel has been closed

### DISCUSSION THREAD

**Discussion by oskarjf** [CoreWCF/CoreWCF#1144] [General]
Hi

I'm upgrading to CoreWCF from WCF and was wondering how it is best to test if the channel has been closed?

The current code is:
OperationContext.Current.Channel.Closed += (sender, args) => ...

I can see that there is a comment on the property in CoreWCF
`        // TODO: Probably want to revert this to public
        internal IContextChannel Channel
        {
            get { return GetCallbackChannel<IContextChannel>(); }
        }
`

**Comment by oskarjf** [comment]
Is this just not possible or?

**Comment by icnocop** [comment]
`GetCallbackChannel` is public, so you can try using this instead:
```
IContextChannel channel = OperationContext.Current.GetCallbackChannel<IContextChannel>();
channel.Closed += (sender, args) => ...
```
